spec2vec/vector_operations.py

```diff
@@ -41,12 +45,14 @@ def _check_model_coverage():
             weights_missing_raised = numpy.power(weights_missing, intensity_weighting_power)
             missing_percentage = 100 * weights_missing_raised.sum() / (weights_raised.sum()
                                                                        + weights_missing_raised.sum())
-            print(f"Found {len(idx_not_in_model)} word(s) missing in the model.",
-                  f"Weighted missing percentage not covered by the given model is {missing_percentage:.2f}%.")
+            msg = (f"Found {len(idx_not_in_model)} word(s) missing in the model.",
+                   f"Weighted missing percentage not covered by the given model is {missing_percentage:.2f}%.")
+            logger.info(msg)
 
-            message = ("Missing percentage is larger than set maximum.",
+            if missing_percentage > allowed_missing_percentage:
+                msg = ("Missing percentage is larger than set maximum.",
                        "Consider retraining the used model or change the `allowed_missing_percentage`.")
-            assert missing_percentage <= allowed_missing_percentage, message
+                logger.warning(msg)
```

Comment:
This looks good.
Additional idea: If above a certain threshold, not a vector is returned, but instead None is returned. 
Other functions taking in the embedding, should than be able to handle None. Probably not worth it, but this would be a practical way of preventing people from using models on spectra that have a lot of missing peaks, without breaking the pipeline for all other spectra in a sample as well. 
The problem with the warning would be: If MS2Query checks for this warning and gets a warning that one of the input spectra is above this threshold, I do not want to discard all results, but only the results of the spectrum that has too many missing words (or add a warning to a single spectrum in the output of MS2Query (instead of an entire sample)). With the current warning system, I would not know how to link the warning to a specific spectrum and therefore will likely not use this information.

Reply:
Good point. That is a design choice and up to us I would say. We could

if > allowed_missing_percentage:
- change nothing (only raise logging warning, that's how it is now in this PR)
- return None 
- return all-zero vector (which would also lead to similarity score of 0)

Reply:
OK, as just discusses online we'll go for the all-zero vectors.